internal/pkg/runtime/engine/singularity/container_linux.go

```diff
@@ -1942,9 +1942,12 @@ func (c *container) addScratchMount(system *mount.System) error {
 
 	workdir := c.engine.EngineConfig.GetWorkdir()
 	hasWorkdir := workdir != ""
-
+	var err error
 	if hasWorkdir {
-		workdir = filepath.Clean(workdir)
+		workdir, err = filepath.Abs(filepath.Clean(workdir))
+		if err != nil {
+			sylog.Warningf("Can't determine absolute path of workdir %s", workdir)
```

Comment:
I believe this needs to be an error return, not a warning. If `filepath.Abs` returns an err, the string it returns into `workdir` will be `""` which will mean we proceed with `workdir` effectively being the cwd. We don't want to have any chance that we create our scratch dir in the cwd, when that's not what the user asked for.

Reply:
Okay - but note that I was just extending what was already present in the native-mode code:

```go
			workdir, err := filepath.Abs(filepath.Clean(workdir))
			if err != nil {
				sylog.Warningf("Can't determine absolute path of workdir %s", workdir)
			}

			tmpSource = filepath.Join(workdir, tmpSource)
			vartmpSource = filepath.Join(workdir, vartmpSource)
```
(internal/pkg/runtime/engine/singularity/container_linux.go:1871)

I'll change it in all four places, then: in {tmp-mounting routine, scratch-mounting routine} x {native code, OCI-mode code}.

Reply:
:+1: Yep, thanks.